<!-------------------------------->
<!--  DO NOT CHANGE STYLE BELOW -->
<style>
H3{color:darkblue !important;font-weight: bold !important}
blockquote{background:lightcyan !important; color:black !important;padding:10pt !important;font-weight: bold !important}
table{font-size:110%  !important}
hr{width: 100% !important; height: 3px !important; background:black !important;}
</style>
<!--  DO NOT CHANGE STYLE ABOVE -->
<!-------------------------------->

---
<h1><center>IDS 435 - Solution Template</center></h1>

    
| First Name        | Last Name     | UIN       | Email                     |
| :---              |    :----:     | :----:    |                   ---:    |
| Suhaib                  |   Bari            | 672206083   |          msuha3@uic.edu                 |
| Yumu                  |   Chen            | 670410366   |          ychen557@uic.edu                |                          |





<!-------------------------------->
<!--  DO NOT CHANGE STYLE BELOW -->
<style>
H3{color:darkblue !important;font-weight: bold !important}
blockquote{background:lightcyan !important; color:black !important;padding:10pt !important;font-weight: bold !important}
table{font-size:110%  !important}
hr{width: 100% !important; height: 3px !important; background:black !important;}
</style>
<!--  DO NOT CHANGE STYLE ABOVE -->
<!-------------------------------->


---

---
### Question 1. Kentwood Electronics

Kentwood electronics is interested in marketing its business in the Chicagoland area.
It is considering three alternatives: TV, radio, and internet.
Kentwood electronics has estimated the number of audience reached per ad, the cost per ad, and the maximum number of ads allowed. This information is reported in the following table:  

|        Component       	    |    Television    	    |    Radio    	    |       Internet    	|
|:--------------------------:	|:--------------------:	|:---------------:	|:------------------:	|
| Audience reached per Ad    	|      100,000         	|    14,000       	|      1        	|
|       Cost per Ad          	|       \$2,000         	|     \$300        	|       \$0.1        	|
|      Maximum usage count     	|         10           	|      20         	|        $\infty$          	|

<br> Your goal is to determine the number of each type of media to use such that the number of audience reached is maximized.
You have to ensure that your solutions adhere to the following constraints:

- The number of each type of media used should not exceed the maximum usage count;
- Total cost of ads should be less than the available budget of $18,000.
- To balance the use of advertising types, television adds may only account for at most 50% of the total budget
- The internet ads should account for at least 10% of the audience.

Please answer the following questions:
### Question 1-1
1. Write down a formulation for this optimization problem. Please define and explain the objective function, decision variables, and the constraints in your formulation.


### **Answer.**

Write your answer to Part (1) of Question 1 in words here.


#### Decision Variables:
x1: The number of television ads to be used.

x2: The number of radio ads to be used.

x3: The number of internet ads to be used.


#### Objective Function:
100,000x1 + 14,000x2 + x3

#### Constraints:
1. **Media Usage Constraints:**
   - Television ads should not exceed 10: \(x1<= 10\)
   - Radio ads should not exceed 20: \(x2 <= 20\)
   - There's no explicit upper limit for Internet ads, but they will be limited by other constraints.

2. **Budget Constraint:**
   The total cost should not exceed $18,000: 2000x1 + 300x2 + 0.1x3 <= 18,000

3. **Television Budget Constraint:**
   The cost of television ads should not exceed 50% of the total budget: 2000x1 <= 9000

4. **Internet Audience Constraint:**
   Internet ads should account for at least 10% of the total audience: x3 >= 0.10 X (100,000x1 + 14,000x2 + x3)



<!-------------------------------->
<!--  DO NOT CHANGE STYLE BELOW -->
<style>
H3{color:darkblue !important;font-weight: bold !important}
blockquote{background:lightcyan !important; color:black !important;padding:10pt !important;font-weight: bold !important}
table{font-size:110%  !important}
hr{width: 100% !important; height: 3px !important; background:black !important;}
</style>
<!--  DO NOT CHANGE STYLE ABOVE -->
<!-------------------------------->


---

### **Question 1-2.**
2. Identify whether your formulation is a linear program, an integer program, or a mixed-integer program.

### **Answer.**

Write your answer to Part (2) of Question 1 in words here. <br></br>
The formulation is a Mixed-Integer Linear Program (MILP)



<!-------------------------------->
<!--  DO NOT CHANGE STYLE BELOW -->
<style>
H3{color:darkblue !important;font-weight: bold !important}
blockquote{background:lightcyan !important; color:black !important;padding:10pt !important;font-weight: bold !important}
table{font-size:110%  !important}
hr{width: 100% !important; height: 3px !important; background:black !important;}
</style>
<!--  DO NOT CHANGE STYLE ABOVE -->
<!-------------------------------->


---

### **Question 1-3.**
3. Write Python code to solve the model using Gurobi.



### **Answer.**

Write your answer to Part (2) of Question 1 in words here. <br></br>


If the question requires using code, then include a code block below.


In [ ]:
!pip install gurobipy

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13.4/13.4 MB 37.0 MB/s eta 0:00:00


In [ ]:
from gurobipy import Model, GRB

m = Model("Kentwood_Electronics")

# Define decision variables
x1 = m.addVar(vtype=GRB.INTEGER, name="x1")  # Number of TV ads
x2 = m.addVar(vtype=GRB.INTEGER, name="x2")  # Number of Radio ads
x3 = m.addVar(vtype=GRB.INTEGER, name="x3")  # Number of Internet ads

# Set the objective function
m.setObjective(100000 * x1 + 14000 * x2 + x3, GRB.MAXIMIZE)

# Add constraints
# Media usage constraints
m.addConstr(x1 <= 10, "TV_ads_limit")
m.addConstr(x2 <= 20, "Radio_ads_limit")

# Budget constraint
m.addConstr(2000 * x1 + 300 * x2 + 0.1 * x3 <= 18000, "Budget_limit")

# TV budget constraint (at most 50% of total budget)
m.addConstr(2000 * x1 <= 9000, "TV_budget_limit")

# Internet audience constraint (at least 10% of total audience)
m.addConstr(x3 >= 0.10 * (100000 * x1 + 14000 * x2 + x3), "Internet_audience_requirement")

# Optimize the model
m.optimize()

# Display the optimal solution
if m.status == GRB.OPTIMAL:
    print(f"Optimal number of TV ads: {x1.X}")
    print(f"Optimal number of Radio ads: {x2.X}")
    print(f"Optimal number of Internet ads: {x3.X}")
    print(f"Maximum audience reached: {m.ObjVal}")


Restricted license - for non-production use only - expires 2025-11-24
Gurobi Optimizer version 11.0.1 build v11.0.1rc0 (win64 - Windows 10.0 (19045.2))

CPU model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz, instruction set [SSE2|AVX|AVX2]
Thread count: 6 physical cores, 12 logical processors, using up to 12 threads

Optimize a model with 5 rows, 3 columns and 9 nonzeros
Model fingerprint: 0x5e260b53
Variable types: 0 continuous, 3 integer (0 binary)
Coefficient statistics:
  Matrix range     [1e-01, 1e+04]
  Objective range  [1e+00, 1e+05]
  Bounds range     [0e+00, 0e+00]
  RHS range        [1e+01, 2e+04]
Found heuristic solution: objective 180000.00000
Presolve removed 4 rows and 1 columns
Presolve time: 0.00s
Presolved: 1 rows, 2 columns, 2 nonzeros
Found heuristic solution: objective 632000.00000
Variable types: 0 continuous, 2 integer (0 binary)

Explored 0 nodes (0 simplex iterations) in 0.02 seconds (0.00 work units)
Thread count was 12 (of 12 available processors)

Solution coun

<!-------------------------------->
<!--  DO NOT CHANGE STYLE BELOW -->
<style>
H3{color:darkblue !important;font-weight: bold !important}
blockquote{background:lightcyan !important; color:black !important;padding:10pt !important;font-weight: bold !important}
table{font-size:110%  !important}
hr{width: 100% !important; height: 3px !important; background:black !important;}
</style>
<!--  DO NOT CHANGE STYLE ABOVE -->
<!-------------------------------->


---

### **Question 1-4.**
4. Report the optimal solution and the optimal value (optimal audience reached) obtained from Gurobi. Also, report the Gurobi runtime.

### **Answer.**

1. **Model Statistics**: The optimization model has 5 rows (constraints), 3 columns (variables), and 9 nonzeros (non-zero coefficients in the constraint matrix).

2. **Variable Types**: All variables are integer, with no continuous or binary variables indicated.

3. **Coefficient Statistics**: Provides the range of values in the constraint matrix, the objective function coefficients, and the right-hand side (RHS) values of the constraints.

4. **Heuristic Solutions**: Gurobi found two heuristic solutions with objective values of 180,000 and 632,000, respectively.

5. **Optimal Solution**: The solver reached an optimal solution with an objective value of 632,000, which represents the maximum audience reached in this context.

6. **Variable Values at Optimality**:
   - TV ads: 4
   - Radio ads: 12
   - Internet ads: 64,000

7. **Runtime**: The optimization process took 0.02 seconds

the optimal advertising strategy, according to the Gurobi Optimizer, involves airing 4 TV ads, 12 radio ads, and 64,000 internet ads to reach a maximum audience of 632,000. The optimization process was completed in 0.02 seconds.



<!-------------------------------->
<!--  DO NOT CHANGE STYLE BELOW -->
<style>
H3{color:darkblue !important;font-weight: bold !important}
blockquote{background:lightcyan !important; color:black !important;padding:10pt !important;font-weight: bold !important}
table{font-size:110%  !important}
hr{width: 100% !important; height: 3px !important; background:black !important;}
</style>
<!--  DO NOT CHANGE STYLE ABOVE -->
<!-------------------------------->


---

### **Question 1-5.**
5. Report the Gurobi output. Look at this output and comment on the following question:

    5.1. Gurobi uses a presolve step to simplify the problem. What is the size of the model after presolve ('rows' being constraints, 'columns' being variable)?

### **Answer.**
Originally, the model had 5 rows (constraints) and 3 columns (variables). After presolve, it was reduced to just 1 row and 2 columns.

- **Rows (constraints)**: The presolve step removed 4 constraints, reducing the total from 5 to 1.
- **Columns (variables)**: The presolve step removed 1 variable, reducing the total from 3 to 2.
- **Nonzeros**: The number of nonzeros in the constraint matrix was reduced to 2, indicating a simpler and more sparse matrix.

The presolve process took virtually no time (0.00 seconds)



<!-------------------------------->
<!--  DO NOT CHANGE STYLE BELOW -->
<style>
H3{color:darkblue !important;font-weight: bold !important}
blockquote{background:lightcyan !important; color:black !important;padding:10pt !important;font-weight: bold !important}
table{font-size:110%  !important}
hr{width: 100% !important; height: 3px !important; background:black !important;}
</style>
<!--  DO NOT CHANGE STYLE ABOVE -->
<!-------------------------------->


---

### **Question 1-6.**

6. **(Graduate Only)** Assume producing TV and Radio ads carries an respective upfront cost of $\$5000$ (TV) and $\$1000$ (Radio). This cost is incurred if at least one ad of the relevant type is aired, but does not increase if more than one ad is run.  

    6.1. Write down a MIP formulation for this modified problem. Be careful not to add any non-linear constraints (besides integrality constraints).

    6.2. Rerun this modified model via Gurobi and report the resulting optimal solution and optimal value.


### **Answer.**
### 6.1 MIP Formulation with Upfront Costs

To incorporate the upfront costs for TV and radio ads into the MIP formulation, we introduce binary variables that indicate whether any ads of a particular type are used.

#### Decision Variables
- \(x1\): The number of television ads.
- \(x2\): The number of radio ads.
- \(x3\): The number of internet ads.
- \(y1\): Binary variable where \(y1 = 1\) if any TV ads are used (\(x1 > 0\)) and \(0\) otherwise.
- \(y2\): Binary variable where \(y2 = 1\) if any Radio ads are used (\(x2 > 0\)) and \(0\) otherwise.

#### Objective Function
Maximize audience reach while considering the upfront costs:
\[ {Maximize} \; 100,000x1 + 14,000x2 + x3 - 5000y1 - 1000y2 \]

#### Constraints
1. Media usage constraints remain the same.
   - \(x1 <= 10\)
   - \(x2 <= 20\)
2. Budget constraint (adjusted for upfront costs):
   - \(2000x1 + 300x2 + 0.1x3 + 5000y1 + 1000y2 <= 18,000\)
3. TV budget constraint (at most 50% of the total budget) remains the same.
   - \(2000x1 + 5000y1 <= 9000\)
4. Internet audience constraint remains the same.
   - \(x3 >= 0.10 X (100,000x1 + 14,000x2 + x3)\)
5. Linking constraints for the binary variables:
   - \(x1 <= 10y1\)
   - \(x2 <= 20y2\)

These linking constraints ensure that y1 and y2 are 1 if there's at least one TV or radio ad, respectively.



In [ ]:
from gurobipy import Model, GRB

m = Model("Kentwood_Electronics_Modified")

# Define decision variables
x1 = m.addVar(vtype=GRB.INTEGER, name="x1")  # Number of TV ads
x2 = m.addVar(vtype=GRB.INTEGER, name="x2")  # Number of Radio ads
x3 = m.addVar(vtype=GRB.INTEGER, name="x3")  # Number of Internet ads
y1 = m.addVar(vtype=GRB.BINARY, name="y1")   # Binary variable for TV ads
y2 = m.addVar(vtype=GRB.BINARY, name="y2")   # Binary variable for Radio ads

# Set the objective function
m.setObjective(100000 * x1 + 14000 * x2 + x3 - 5000 * y1 - 1000 * y2, GRB.MAXIMIZE)

# Add constraints
m.addConstr(x1 <= 10, "TV_ads_limit")
m.addConstr(x2 <= 20, "Radio_ads_limit")
m.addConstr(2000 * x1 + 300 * x2 + 0.1 * x3 + 5000 * y1 + 1000 * y2 <= 18000, "Budget_limit")
m.addConstr(2000 * x1 + 5000 * y1 <= 9000, "TV_budget_limit")
m.addConstr(x3 >= 0.10 * (100000 * x1 + 14000 * x2 + x3), "Internet_audience_requirement")
m.addConstr(x1 <= 10 * y1, "Linking_constraint_TV")
m.addConstr(x2 <= 20 * y2, "Linking_constraint_Radio")

m.optimize()

if m.status == GRB.OPTIMAL:
    print(f"Optimal number of TV ads: {x1.X}")
    print(f"Optimal number of Radio ads: {x2.X}")
    print(f"Optimal number of Internet ads: {x3.X}")
    print(f"Maximum audience reached: {m.ObjVal}")


Gurobi Optimizer version 11.0.1 build v11.0.1rc0 (win64 - Windows 10.0 (19045.2))

CPU model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz, instruction set [SSE2|AVX|AVX2]
Thread count: 6 physical cores, 12 logical processors, using up to 12 threads

Optimize a model with 7 rows, 5 columns and 16 nonzeros
Model fingerprint: 0xd85a3dab
Variable types: 0 continuous, 5 integer (2 binary)
Coefficient statistics:
  Matrix range     [1e-01, 1e+04]
  Objective range  [1e+00, 1e+05]
  Bounds range     [1e+00, 1e+00]
  RHS range        [1e+01, 2e+04]
Found heuristic solution: objective 180000.00000
Presolve removed 5 rows and 2 columns
Presolve time: 0.00s
Presolved: 2 rows, 3 columns, 5 nonzeros
Found heuristic solution: objective 389000.00000
Variable types: 0 continuous, 3 integer (1 binary)

Root relaxation: objective 4.135122e+05, 1 iterations, 0.00 seconds (0.00 work units)

    Nodes    |    Current Node    |     Objective Bounds      |     Work
 Expl Unexpl |  Obj  Depth IntInf | Incumbent 

This output shows that to maximize the audience reached, the optimal advertising strategy is to run 2 TV ads, 12 radio ads, and 44,000 internet ads, resulting in a maximum audience of 406,000.

<!-------------------------------->
<!--  DO NOT CHANGE STYLE BELOW -->
<style>
H3{color:darkblue !important;font-weight: bold !important}
blockquote{background:lightcyan !important; color:black !important;padding:10pt !important;font-weight: bold !important}
table{font-size:110%  !important}
hr{width: 100% !important; height: 3px !important; background:black !important;}
</style>
<!--  DO NOT CHANGE STYLE ABOVE -->
<!-------------------------------->


---

### **Question 1-7.**
7. **(Bonus for Graduate and Undergraduate)** Rerun your model from 3. (i.e., without the graduate only modification) to answer the following questions: By how much does the optimal value change if

    7.1. the budget is increased by 5%?
    
    7.2. the maximum usage count of  television reduces by 10%?

    7.3 the integer constraints are relaxed (i.e., effectively allowing shorter, partial ads)?

### **Answer.**

Write your answer to Part (2) of Question 1 in words here. <br></br>


If the question requires using code, then include a code block below.


In [ ]:
#7.1 the budget is increased by 5%
from gurobipy import Model, GRB

m = Model("Kentwood_Electronics")

# Define decision variables
x1 = m.addVar(vtype=GRB.INTEGER, name="x1")  # Number of TV ads
x2 = m.addVar(vtype=GRB.INTEGER, name="x2")  # Number of Radio ads
x3 = m.addVar(vtype=GRB.INTEGER, name="x3")  # Number of Internet ads

# Set the objective function
m.setObjective(100000 * x1 + 14000 * x2 + x3, GRB.MAXIMIZE)

# Add constraints
# Media usage constraints
m.addConstr(x1 <= 10, "TV_ads_limit")
m.addConstr(x2 <= 20, "Radio_ads_limit")

# Adjusted budget constraint the budget is increased by 5% 18000*1.05=18900
m.addConstr(2000 * x1 + 300 * x2 + 0.1 * x3 <= 18900, "Adjusted_Budget_limit")


# TV budget constraint (at most 50% of total budget)
m.addConstr(2000 * x1 <= 9000, "TV_budget_limit")

# Internet audience constraint (at least 10% of total audience)
m.addConstr(x3 >= 0.10 * (100000 * x1 + 14000 * x2 + x3), "Internet_audience_requirement")

# Optimize the model
m.optimize()

# Display the optimal solution
if m.status == GRB.OPTIMAL:
    print(f"Optimal number of TV ads: {x1.X}")
    print(f"Optimal number of Radio ads: {x2.X}")
    print(f"Optimal number of Internet ads: {x3.X}")
    print(f"Maximum audience reached: {m.ObjVal}")


Restricted license - for non-production use only - expires 2025-11-24
Gurobi Optimizer version 11.0.1 build v11.0.1rc0 (win64 - Windows 10.0 (19045.2))

CPU model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz, instruction set [SSE2|AVX|AVX2]
Thread count: 6 physical cores, 12 logical processors, using up to 12 threads

Optimize a model with 5 rows, 3 columns and 9 nonzeros
Model fingerprint: 0xf7a3fdd2
Variable types: 0 continuous, 3 integer (0 binary)
Coefficient statistics:
  Matrix range     [1e-01, 1e+04]
  Objective range  [1e+00, 1e+05]
  Bounds range     [0e+00, 0e+00]
  RHS range        [1e+01, 2e+04]
Found heuristic solution: objective 189000.00000
Presolve removed 5 rows and 3 columns
Presolve time: 0.07s
Presolve: All rows and columns removed

Explored 0 nodes (0 simplex iterations) in 0.19 seconds (0.00 work units)
Thread count was 1 (of 12 available processors)

Solution count 2: 663000 189000 

Optimal solution found (tolerance 1.00e-04)
Best objective 6.630000000000e+05, bes

The original optimal value for the maximum audience reached was 632,000. After increasing the budget by 5%, the new optimal value is 663,000.

The increase in the maximum audience reached is:

 663,000 - 632,000 = 31,000

This shows that with a 5% increase in the budget, the maximum audience reached increased by 31,000. The change in the advertising strategy to achieve this new optimal value includes the same number of TV ads (4), an increase in radio ads from 12 to 14, and an increase in internet ads from 64,000 to 67,000.

 7.2. the maximum usage count of  television reduces by 10%?

In [ ]:
# 7.2. the maximum usage count of television reduces by 10%?
from gurobipy import Model, GRB

m = Model("Kentwood_Electronics")

# Define decision variables
x1 = m.addVar(vtype=GRB.INTEGER, name="x1")  # Number of TV ads
x2 = m.addVar(vtype=GRB.INTEGER, name="x2")  # Number of Radio ads
x3 = m.addVar(vtype=GRB.INTEGER, name="x3")  # Number of Internet ads

# Set the objective function
m.setObjective(100000 * x1 + 14000 * x2 + x3, GRB.MAXIMIZE)

# Adjusted TV ads constraint
m.addConstr(x1 <= 9, "Adjusted_TV_ads_limit")  #  the maximum usage count of television reduces by 10%? 10 * 0.90 = 9
m.addConstr(x2 <= 20, "Radio_ads_limit")

# budget constraint
m.addConstr(2000 * x1 + 300 * x2 + 0.1 * x3 <= 18000, "Budget_limit")


# TV budget constraint (at most 50% of total budget)
m.addConstr(2000 * x1 <= 9000, "TV_budget_limit")

# Internet audience constraint (at least 10% of total audience)
m.addConstr(x3 >= 0.10 * (100000 * x1 + 14000 * x2 + x3), "Internet_audience_requirement")

# Optimize the model
m.optimize()

# Display the optimal solution
if m.status == GRB.OPTIMAL:
    print(f"Optimal number of TV ads: {x1.X}")
    print(f"Optimal number of Radio ads: {x2.X}")
    print(f"Optimal number of Internet ads: {x3.X}")
    print(f"Maximum audience reached: {m.ObjVal}")


Gurobi Optimizer version 11.0.1 build v11.0.1rc0 (linux64 - "Ubuntu 22.04.3 LTS")

CPU model: Intel(R) Xeon(R) CPU @ 2.20GHz, instruction set [SSE2|AVX|AVX2]
Thread count: 1 physical cores, 2 logical processors, using up to 2 threads

Optimize a model with 5 rows, 3 columns and 9 nonzeros
Model fingerprint: 0x76ceee07
Variable types: 0 continuous, 3 integer (0 binary)
Coefficient statistics:
  Matrix range     [1e-01, 1e+04]
  Objective range  [1e+00, 1e+05]
  Bounds range     [0e+00, 0e+00]
  RHS range        [9e+00, 2e+04]
Found heuristic solution: objective 180000.00000
Presolve removed 4 rows and 1 columns
Presolve time: 0.00s
Presolved: 1 rows, 2 columns, 2 nonzeros
Found heuristic solution: objective 632000.00000
Variable types: 0 continuous, 2 integer (0 binary)

Explored 0 nodes (0 simplex iterations) in 0.02 seconds (0.00 work units)
Thread count was 2 (of 2 available processors)

Solution count 2: 632000 180000 

Optimal solution found (tolerance 1.00e-04)
Best objective 6.32


When we reduce the maximum number of TV ads by 10%, lowering it from 10 to 9, this new limit still doesn't affect the original optimal solution because it only utilized 4 TV ad slots, which is below the new limit of 9.

7.3 the integer constraints are relaxed (i.e., effectively allowing shorter, partial ads)?

In [ ]:
from gurobipy import Model, GRB

m = Model("Kentwood_Electronics")

# Define decision variables

x1 = m.addVar(vtype=GRB.CONTINUOUS, name="x1")
x2 = m.addVar(vtype=GRB.CONTINUOUS, name="x2")
x3 = m.addVar(vtype=GRB.CONTINUOUS, name="x3")

# Set the objective function
m.setObjective(100000 * x1 + 14000 * x2 + x3, GRB.MAXIMIZE)

# Add constraints
# Media usage constraints
m.addConstr(x1 <= 10, "TV_ads_limit")
m.addConstr(x2 <= 20, "Radio_ads_limit")

# Adjusted budget constraint
m.addConstr(2000 * x1 + 300 * x2 + 0.1 * x3 <= 18000, "Adjusted_Budget_limit")


# TV budget constraint (at most 50% of total budget)
m.addConstr(2000 * x1 <= 9000, "TV_budget_limit")

# Internet audience constraint (at least 10% of total audience)
m.addConstr(x3 >= 0.10 * (100000 * x1 + 14000 * x2 + x3), "Internet_audience_requirement")

# Optimize the model
m.optimize()

# Display the optimal solution
if m.status == GRB.OPTIMAL:
    print(f"Optimal number of TV ads: {x1.X}")
    print(f"Optimal number of Radio ads: {x2.X}")
    print(f"Optimal number of Internet ads: {x3.X}")
    print(f"Maximum audience reached: {m.ObjVal}")


Gurobi Optimizer version 11.0.1 build v11.0.1rc0 (win64 - Windows 10.0 (19045.2))

CPU model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz, instruction set [SSE2|AVX|AVX2]
Thread count: 6 physical cores, 12 logical processors, using up to 12 threads

Optimize a model with 5 rows, 3 columns and 9 nonzeros
Model fingerprint: 0x20bded8c
Coefficient statistics:
  Matrix range     [1e-01, 1e+04]
  Objective range  [1e+00, 1e+05]
  Bounds range     [0e+00, 0e+00]
  RHS range        [1e+01, 2e+04]
Presolve removed 4 rows and 1 columns
Presolve time: 0.01s
Presolved: 1 rows, 2 columns, 2 nonzeros

Iteration    Objective       Primal Inf.    Dual Inf.      Time
       0    6.4285714e+05   3.291429e+02   0.000000e+00      0s
       1    6.3658537e+05   0.000000e+00   0.000000e+00      0s

Solved in 1 iterations and 0.02 seconds (0.00 work units)
Optimal objective  6.365853659e+05
Optimal number of TV ads: 4.5
Optimal number of Radio ads: 8.78048780487805
Optimal number of Internet ads: 63658.5365853

The output from Gurobi after relaxing the integer constraints shows a new optimal solution where the decision variables, representing the number of ads, can take on fractional values.

Comparing this to the previous integer-constrained solution:

- The optimal number of TV ads increased from 4 to 4.5.
- The optimal number of Radio ads changed from 12 to approximately 8.78, showing a decrease.
- The optimal number of Internet ads has slightly decreased from 64,000 to approximately 63,658.54.
- The maximum audience reached has increased from 632,000 to approximately 636,585.36.